### PR TITLE
Documentation for security mitigation

### PR DESCRIPTION
**Pull request by jianrongzhang89**
## Description
Update the docs related to the security mitigation.

## Which issue(s) does this PR fix or relate to

- Fixes #160 

## PR acceptance criteria

- [ ] Tests
- [ X] Documentation

## How to test changes / Special notes to the reviewer
None

### CODE REVIEW

File: docs/admin.md [hunk 1]
```diff
@@ -81,6 +81,19 @@ kubectl get -n backstage-system configmap default-config > my-config.yaml
 
 It has to be re-applied to the controller's container after being reconciled by kubernetes processes.
 
+### Recommended Namespace for Operator Installation
+It is recommended to deploy the Backstage Operator in a dedicated default namespace `backstage-system`. This allows cluster administrators to restrict that the operator gets installed on particular namespaces. For instance, the cluster admininistrator can require that the operator to only get installed on `backstage-system` by creating the following `OperatorGroup`:
```
Comment (rm3l) [inline docs/admin.md:85]:
```suggestion
It is recommended to deploy the Backstage Operator in a dedicated default namespace `backstage-system`. This allows cluster administrators to restrict that the operator gets installed on particular namespaces. For instance, the cluster administrator can require that the operator only gets installed on `backstage-system` by creating the following [`OperatorGroup`](https://olm.operatorframework.io/docs/concepts/crds/operatorgroup/):
```

File: docs/admin.md [hunk 2]
```diff
@@ -96,4 +109,8 @@ By default, the Backstage Operator is configured to use publicly available image
 If you plan to deploy to a [restricted environment](https://docs.openshift.com/container-platform/4.14/operators/admin/olm-restricted-networks.html),
 you will need to configure your cluster or network to allow these images to be pulled.
 For the list of related images deployed by the Operator, see the `RELATED_IMAGE_*` env vars or `relatedImages` section of the [CSV](../bundle/manifests/backstage-operator.clusterserviceversion.yaml).
-See also https://docs.openshift.com/container-platform/4.14/operators/admin/olm-restricted-networks.html
\ No newline at end of file
+See also https://docs.openshift.com/container-platform/4.14/operators/admin/olm-restricted-networks.html
+
+#### Custom Backstage Image
+
+You can use the Backstage Operator to deploy a backstage application with your custom backstage image by setting the field `spec.application.image` in your Backstage CR. This is at your own risk and it is your responsibility to ensure that the image is from trusted sources, has been tested and validated for security complinance.
```
Comment (rm3l) [inline docs/admin.md:116]:
```suggestion
You can use the Backstage Operator to deploy a backstage application with your custom backstage image by setting the field `spec.application.image` in your Backstage CR. This is at your own risk and it is your responsibility to ensure that the image is from trusted sources, and has been tested and validated for security compliance.
```

**Review by rm3l — CHANGES_REQUESTED** [review]

File: docs/admin.md [hunk 3]
```diff
@@ -81,6 +81,19 @@ kubectl get -n backstage-system configmap default-config > my-config.yaml
...
+apiVersion: operators.coreos.com/v1
+kind: OperatorGroup
+metadata:
+  name: backstage-operator-group
+  namespace: backstage-system
+spec:
+  targetNamespaces:
+  - backstage-system
+```
+You can choose to deploy the operator in the `openshift-operators` namespace instead. However, you should keep in mind that the Backstage Operator shares the namespace with other operators and therefore any users who can create workloads in that namespace can get their privileges escalated from all operators' service accounts.
```
Comment (rm3l) [inline docs/admin.md:96]:
```suggestion
On OpenShift, you can choose to deploy the operator in the `openshift-operators` namespace instead. However, you should keep in mind that the Backstage Operator shares the namespace with other operators, and therefore any users who can create workloads in that namespace can get their privileges escalated from all operators' service accounts.
```

**Review by nickboldt — CHANGES_REQUESTED** [review]
see comment about default namespace

**Comment by jianrongzhang89** [comment]
> We have a problem with the current recommended namespace for RHDH operator install:
> 
> [1f8875b#diff-b62e6ae7fdd151e3ffaea0cfd39b32492fba5ada80a04f64d7646282ac8e39f2R28](https://github.com/janus-idp/operator/commit/1f8875b1b83e20ed708eea26b1b5647975dd1ae0#diff-b62e6ae7fdd151e3ffaea0cfd39b32492fba5ada80a04f64d7646282ac8e39f2R28)
> 
> `forbidden: cannot request a project starting with "openshift-"`
> 
> So while in these docs you use `backstage-system` we would either need to use `rhdh-system` or revert to the default `openshift-operators` namespace.
> 
> Though TBH I'm baffled why we want "system" instead of "operator" since that's the only thing in that namespace, right? all the other pods (openshift4/ose-kube-rbac-proxy, rhdh, postgres) would be in `mybackstage` or `myrhdhinstancefoo` namespace

There are two different conventions for naming namespaces for operators. In k8s, ***-system are widely used, while in OpenShift, ***-operator are commonly  used. So for downstream, I think rhdh-operator is a good name, while for upstream/mid-stream,  backstage-system is a good choice too.

File: docs/admin.md [hunk 4]
```diff
@@ -96,4 +98,8 @@ By default, the Backstage Operator is configured to use publicly available image
 If you plan to deploy to a [restricted environment](https://docs.openshift.com/container-platform/4.14/operators/admin/olm-restricted-networks.html),
 you will need to configure your cluster or network to allow these images to be pulled.
 For the list of related images deployed by the Operator, see the `RELATED_IMAGE_*` env vars or `relatedImages` section of the [CSV](../bundle/manifests/backstage-operator.clusterserviceversion.yaml).
-See also https://docs.openshift.com/container-platform/4.14/operators/admin/olm-restricted-networks.html
\ No newline at end of file
+See also https://docs.openshift.com/container-platform/4.14/operators/admin/olm-restricted-networks.html
+
+#### Custom Backstage Image
+
+On You can use the Backstage Operator to deploy a backstage application with your custom backstage image by setting the field `spec.application.image` in your Backstage CR. This is at your own risk and it is your responsibility to ensure that the image is from trusted sources, and has been tested and validated for security compliance.
```
Comment (rm3l) [inline docs/admin.md:105]:
```suggestion
You can use the Backstage Operator to deploy a backstage application with your custom backstage image by setting the field `spec.application.image` in your Backstage CR. This is at your own risk and it is your responsibility to ensure that the image is from trusted sources, and has been tested and validated for security compliance.
```

**Review by rm3l — APPROVED** [review]
/hold

@nickboldt Suggested namespace is now `rhdh-operator`, which works with our [install script](https://github.com/janus-idp/operator/blob/main/.rhdh/scripts/install-rhdh-catalog-source.sh). Is that okay for you to merge this PR?

**Comment by rm3l** [comment]
> /hold
> 
> @nickboldt Suggested namespace is now `rhdh-operator`, which works with our [install script](https://github.com/janus-idp/operator/blob/main/.rhdh/scripts/install-rhdh-catalog-source.sh?rgh-link-date=2024-02-07T14%3A30%3A19Z). Is that okay for you to merge this PR?

/hold cancel

Merging this to unblock the doc part.

**Pull request merged**